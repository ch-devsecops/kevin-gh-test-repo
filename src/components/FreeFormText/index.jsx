import React from 'react';
import styled from 'styled-components';
import css from '@styled-system/css';
import { Box, MarkdownHeading, Markdown, H6 } from '@honda-canada/design-system-react';
import { Text } from '@sitecore-jss/sitecore-jss-react';
import { mapGTMCategory } from '../../utils/sitecoreFields';
import { parseBool } from '../../utils/common';

const StyledMarkdown = styled(Markdown)(
  css({
    // Overrides some opinionated CSS from the Acura current-state header
    sup: {
      lineHeight: 'inherit',
    },
  }),
);

// eslint-disable-next-line react/display-name
export default function ({ fields, params, rendering }) {
  if (!fields || !fields.contentBody || !fields.contentBody.value) {
    return null;
  }

  return (
    <Box
      id={fields.anchorId?.value}
      textAlign={params?.contentAlignment?.toLowerCase()}
      width={parseBool(params?.isReducedWidth) ? [1, 3 / 4] : 'initial'}
      mb={rendering?.params?.bottomMargin ? `${rendering?.params?.bottomMargin}px` : ['xl', 'xxl']}
      mt={rendering?.params?.topMargin ? `${rendering?.params?.topMargin}px` : 'zero'}
      data-gtm-component-type={rendering?.componentName}
      data-gtm-category={mapGTMCategory(fields.gtmCategory)}
    >
      {fields.contentTitle?.value && (
        <Box mb={['default', 'l']}>
          <MarkdownHeading headingOverride="h3">{fields.contentTitle?.value}</MarkdownHeading>
        </Box>
      )}
      <Box>
        {fields.subtitle?.value && (
          <H6 as={fields.seoH1?.value === 'Subtitle' && 'h1'}>
            <Text field={fields.subtitle} />
          </H6>
        )}
        <StyledMarkdown
          options={
            fields.seoH1?.value === 'Description' && {
              overrides: {
                p: {
                  component: ({ children, type, ...props }) => {
                    const Elem = type;
                    return <Elem {...props}>{children}</Elem>;
                  },
                  props: {
                    className: 'override',
                    type: 'h1',
                  },
                },
              },
            }
          }
        >
          {fields.contentBody.value}
        </StyledMarkdown>
      </Box>
    </Box>
  );
}
