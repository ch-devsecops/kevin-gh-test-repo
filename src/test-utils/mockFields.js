/* eslint-disable max-len */
/* eslint-disable max-classes-per-file */

// FIXME: jsdoc popup not working on test files
/** Class representing a mock data. */
class MockData {
  /**
   * Add a field.
   * @param {string} key - key of the field obj: i.e. 'title'
   * @param {Object} value - value of the field obj: i.e. { value: 'some text' }.
   */
  add(key, value) {
    this[key] = value;
  }
}

/**
 * Class representing mock fields object.
 * @extends MockData
 *
 * available method: add()
 *
 * default fields: [anchorId, title, bodyText, image, ctaLink, ctaType, ctaIcon, gtmCategory, gtmTitle, contentTitle, subtitle]
 *
 * note: we can add more commonly used fields in the class declaration if needed
 */
class MockFields extends MockData {
  constructor() {
    super();

    this.title = { value: 'Title' };
    this.bodyText = { value: 'BodyText' };
    this.image = {
      value: { src: 'mock.png', alt: 'mock alt' },
    };
    this.ctaLink = {
      value: {
        href: 'www.source.ca',
        title: 'cta aria label',
        text: 'CTA Text',
      },
    };
    this.ctaType = {
      value: 'Tertiary',
    };
    this.ctaIcon = {
      fields: {
        value: {
          value: 'arrowRight',
        },
      },
    };
    this.gtmCategory = {
      fields: {
        value: {
          value: 'Others',
        },
      },
    };
    this.gtmTitle = {
      value: 'gtm title',
    };
    this.contentTitle = {
      value: 'Content Title',
    };
    this.subtitle = {
      value: 'Content Subtitle',
    };
    this.seoH1 = {
      value: '',
    };
    this.anchorId = {
      value: 'anchor-id',
    };
    this.ctaLabel = {
      value: 'CTA Label',
    };
  }
}

/**
 * Class representing a mock rendering object
 * @extends MockData
 *
 * available method: add()
 */
class MockRendering extends MockData {
  /**
   * Create a mock rendering object
   * @param {string} componentName - adds componentName
   */
  constructor(componentName) {
    super();

    this.componentName = componentName;
  }
}

export { MockFields, MockRendering };
