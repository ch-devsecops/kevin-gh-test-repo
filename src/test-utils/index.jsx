import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { SitecoreContext } from '@sitecore-jss/sitecore-jss-react';
import { BrowserRouter } from 'react-router-dom';
import { OptionalVideoModalContextProvider } from '@honda-canada/design-system-react';
import UserLocationProvider from '@honda-canada/user-location';
import { configureAxe } from 'jest-axe';
import ModelExplorationContextProvider from '../components/ModelExplorationContext';
import ModelFiltersContextProvider from '../components/ModelFiltersContext';
import { MockFields, MockRendering } from './mockFields';
import MockTheme from './mockTheme';
import 'intersection-observer';
import { ACURA_THEME_NAME } from '../utils/constants';

const generateJssWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <SitecoreContext
        context={{
          language: 'en',
          site: { name: ACURA_THEME_NAME },
          ...context?.sitecoreContext,
        }}
        componentFactory={{}}
      >
        <BrowserRouter>
          <UserLocationProvider>
            <ThemeProvider theme={context?.theme || MockTheme}>
              <OptionalVideoModalContextProvider>
                <ModelExplorationContextProvider>
                  <ModelFiltersContextProvider>{children}</ModelFiltersContextProvider>
                </ModelExplorationContextProvider>
              </OptionalVideoModalContextProvider>
            </ThemeProvider>
          </UserLocationProvider>
        </BrowserRouter>
      </SitecoreContext>
    );
  };
/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in an typical JSS app's context providers.
 * Note that you can include a `context` object to set or override values
 * for SitecoreContext and ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.context] - default values for SitecoreContext, ThemeProvider
 * @param {string} [options.context.sitecoreContext] - value for SitecoreContext
 * @param {string} [options.context.theme] - value for ThemeProvider
 * @example
 * const { getByText } = jssRender(
 *  <MyComponent />, { context: { sitecoreContext: { language: 'fr' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const jssRender = (component, options = {}) => {
  const { context } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.context;

  return render(component, {
    wrapper: generateJssWrapper(context),
    ...options,
  });
};

const generateWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <BrowserRouter>
        <ThemeProvider theme={context?.theme || MockTheme}>
          <OptionalVideoModalContextProvider>{children}</OptionalVideoModalContextProvider>
        </ThemeProvider>
      </BrowserRouter>
    );
  };

/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in typical design-system context providers.
 * Note that you can include a `theme` object to set or override values
 * for ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.theme] - default value for ThemeProvider
 * @example
 * const { getByText } = render(
 *  <MyComponent />, { theme: { space: { wrapperGap: '15px' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const customRender = (component, options = {}) => {
  const { theme } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.theme;

  return render(component, {
    wrapper: generateWrapper(theme),
    ...options,
  });
};

// disable accessibility checks for errors that are usually trigged by content mistakes
const axe = configureAxe({
  rules: {
    'duplicate-id': { enabled: false },
    'heading-order': { enabled: false },
  },
});

export * from 'jest-axe';
export { axe };
export * from '@testing-library/react';
export { customRender as render, jssRender };
export { MockFields, MockRendering, MockTheme };
