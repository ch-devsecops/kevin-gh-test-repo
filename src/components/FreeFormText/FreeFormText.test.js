import React from 'react';
import { jssRender } from 'test-utils';
import FreeFormText from './index';

const routeData = require('../../../sample-mock-data/routes/free-form-text/en.json');

const props = routeData.placeholders['jss-main'][0];

describe('<FreeFormText>', () => {
  it('should render without errors', () => {
    const { container } = jssRender(<FreeFormText {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('should render with center-aligned content', () => {
    const { container } = jssRender(<FreeFormText {...props} params={{ contentAlignment: 'Center' }} />);

    expect(container).toMatchSnapshot();
  });

  it('should render with an seoH1 value of "Description"', () => {
    const { container } = jssRender(
      <FreeFormText {...props} fields={{ ...props.fields, seoH1: { value: 'Description' } }} />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render with an seoH1 value of "Subtitle"', () => {
    const { container } = jssRender(
      <FreeFormText {...props} fields={{ ...props.fields, seoH1: { value: 'Subtitle' } }} />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should apply topMargin and bottomMargin params', () => {
    const { container } = jssRender(<FreeFormText {...props} params={{ topMargin: 96, bottomMargin: 10 }} />);

    expect(container).toMatchSnapshot();
  });

  it('should apply 3/4 width params', () => {
    const { container } = jssRender(<FreeFormText {...props} params={{ isReducedWidth: true }} />);

    expect(container).toMatchSnapshot();
  });
});
