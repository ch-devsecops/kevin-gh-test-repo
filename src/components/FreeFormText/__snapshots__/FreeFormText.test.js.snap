// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FreeFormText> should apply 3/4 width params 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 UcmHr"
    data-gtm-category="Others"
    width="1,0.75"
  >
    <div
      class="Box-sc-1bsf784-0 ezzCAj"
    >
      <div
        class="sc-1k44dfv-0 klHKZs"
      >
        

        <h3
          class="sc-1k44dfv-0 eUGlGC"
        >
          Title
        </h3>
      </div>
    </div>
    <div
      class="Box-sc-1bsf784-0 iPRsIc"
    >
      <h6
        class="sc-1k44dfv-0 eeeuqx"
      >
        Subtitle
      </h6>
      <div
        class="sc-1k44dfv-0 klHKZs sc-bdVaJa bIYKxT"
      >
        

        <p>
          Lorem ip sum 
          <sup>
            dolor sit
          </sup>
           amet, consectetur adipiscing elit. Nunc scelerisque ex at felis condimentum imperdiet. Proin sodales elit vitae est laoreet, eget vulputate quam pretium. Nulla scelerisque ligula at malesuada mattis. Nulla vulputate posuere orci, id tristique eros pretium nec. Sed a maximus tellus. Ut vel ullamcorper tellus. Nulla ut mi vitae nunc pellentesque mattis. 
        </p>
        <blockquote>
          <p>
            Good-looking, powerful, and a sharp handler, it proves you shouldn’t send a crossover out to do a sports sedan’s job.
          </p>
        </blockquote>
        <h6
          id="unordered-list"
        >
          Unordered List
        </h6>
        <ul>
          <li>
            You can have lists, like this one
          </li>
          <li>
            Make things 
            <strong>
              bold
            </strong>
             or 
            <em>
              italic
            </em>
          </li>
          <li>
            Embed snippets of 
            <code>
              code
            </code>
          </li>
          <li>
            Create 
            <a
              href="/"
            >
              links
            </a>
          </li>
          <li>
            ...
          </li>
        </ul>
        <h6
          id="ordered-list"
        >
          Ordered List
        </h6>
        <ol
          start="1"
        >
          <li>
            You can have lists, like this one
          </li>
          <li>
            Make things 
            <strong>
              bold
            </strong>
             or 
            <em>
              italic
            </em>
          </li>
          <li>
            Embed snippets of 
            <code>
              code
            </code>
          </li>
          <li>
            Create 
            <a
              href="/"
            >
              links
            </a>
          </li>
          <li>
            ...
          </li>
        </ol>
        <h6
          id="another-sub-heading"
        >
          Another Sub Heading
        </h6>
        <p>
          Lorem ip sum dolor sit amet, consectetur adipiscing elit. Nunc scelerisque ex at felis condimentum imperdiet. Proin sodales elit vitae est laoreet, eget vulputate quam pretium. Nulla scelerisque ligula at malesuada mattis. Nulla vulputate posuere orci, id tristique eros pretium nec. Sed a maximus tellus. Ut vel ullamcorper tellus. Nulla ut mi vitae nunc pellentesque mattis.
        </p>
        <h6
          id="another-sub-heading"
        >
          Another Sub Heading
        </h6>
        <p>
          Lorem ip sum dolor sit amet, consectetur adipiscing elit. Nunc scelerisque ex at felis condimentum imperdiet. Proin sodales elit vitae est laoreet, eget vulputate quam pretium. Nulla scelerisque ligula at malesuada mattis. Nulla vulputate posuere orci, id tristique eros pretium nec. Sed a maximus tellus. Ut vel ullamcorper tellus. Nulla ut mi vitae nunc pellentesque mattis.
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<FreeFormText> should apply topMargin and bottomMargin params 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 gdTPBx"
    data-gtm-category="Others"
    width="initial"
  >
    <div
      class="Box-sc-1bsf784-0 ezzCAj"
    >
      <div
        class="sc-1k44dfv-0 klHKZs"
      >
        

        <h3
          class="sc-1k44dfv-0 eUGlGC"
        >
          Title
        </h3>
      </div>
    </div>
    <div
      class="Box-sc-1bsf784-0 iPRsIc"
    >
      <h6
        class="sc-1k44dfv-0 eeeuqx"
      >
        Subtitle
      </h6>
      <div
        class="sc-1k44dfv-0 klHKZs sc-bdVaJa bIYKxT"
      >
        

        <p>
          Lorem ip sum 
          <sup>
            dolor sit
          </sup>
           amet, consectetur adipiscing elit. Nunc scelerisque ex at felis condimentum imperdiet. Proin sodales elit vitae est laoreet, eget vulputate quam pretium. Nulla scelerisque ligula at malesuada mattis. Nulla vulputate posuere orci, id tristique eros pretium nec. Sed a maximus tellus. Ut vel ullamcorper tellus. Nulla ut mi vitae nunc pellentesque mattis. 
        </p>
        <blockquote>
          <p>
            Good-looking, powerful, and a sharp handler, it proves you shouldn’t send a crossover out to do a sports sedan’s job.
          </p>
        </blockquote>
        <h6
          id="unordered-list"
        >
          Unordered List
        </h6>
        <ul>
          <li>
            You can have lists, like this one
          </li>
          <li>
            Make things 
            <strong>
              bold
            </strong>
             or 
            <em>
              italic
            </em>
          </li>
          <li>
            Embed snippets of 
            <code>
              code
            </code>
          </li>
          <li>
            Create 
            <a
              href="/"
            >
              links
            </a>
          </li>
          <li>
            ...
          </li>
        </ul>
        <h6
          id="ordered-list"
        >
          Ordered List
        </h6>
        <ol
          start="1"
        >
          <li>
            You can have lists, like this one
          </li>
          <li>
            Make things 
            <strong>
              bold
            </strong>
             or 
            <em>
              italic
            </em>
          </li>
          <li>
            Embed snippets of 
            <code>
              code
            </code>
          </li>
          <li>
            Create 
            <a
              href="/"
            >
              links
            </a>
          </li>
          <li>
            ...
          </li>
        </ol>
        <h6
          id="another-sub-heading"
        >
          Another Sub Heading
        </h6>
        <p>
          Lorem ip sum dolor sit amet, consectetur adipiscing elit. Nunc scelerisque ex at felis condimentum imperdiet. Proin sodales elit vitae est laoreet, eget vulputate quam pretium. Nulla scelerisque ligula at malesuada mattis. Nulla vulputate posuere orci, id tristique eros pretium nec. Sed a maximus tellus. Ut vel ullamcorper tellus. Nulla ut mi vitae nunc pellentesque mattis.
        </p>
        <h6
          id="another-sub-heading"
        >
          Another Sub Heading
        </h6>
        <p>
          Lorem ip sum dolor sit amet, consectetur adipiscing elit. Nunc scelerisque ex at felis condimentum imperdiet. Proin sodales elit vitae est laoreet, eget vulputate quam pretium. Nulla scelerisque ligula at malesuada mattis. Nulla vulputate posuere orci, id tristique eros pretium nec. Sed a maximus tellus. Ut vel ullamcorper tellus. Nulla ut mi vitae nunc pellentesque mattis.
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<FreeFormText> should render with an seoH1 value of "Description" 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 gdTPBx"
    data-gtm-category="Others"
    width="initial"
  >
    <div
      class="Box-sc-1bsf784-0 ezzCAj"
    >
      <div
        class="sc-1k44dfv-0 klHKZs"
      >
        

        <h3
          class="sc-1k44dfv-0 eUGlGC"
        >
          Title
        </h3>
      </div>
    </div>
    <div
      class="Box-sc-1bsf784-0 iPRsIc"
    >
      <h6
        class="sc-1k44dfv-0 eeeuqx"
      >
        Subtitle
      </h6>
      <div
        class="sc-1k44dfv-0 klHKZs sc-bdVaJa bIYKxT"
      >
        

        <h1
          class="override"
        >
          Lorem ip sum 
          <sup>
            dolor sit
          </sup>
           amet, consectetur adipiscing elit. Nunc scelerisque ex at felis condimentum imperdiet. Proin sodales elit vitae est laoreet, eget vulputate quam pretium. Nulla scelerisque ligula at malesuada mattis. Nulla vulputate posuere orci, id tristique eros pretium nec. Sed a maximus tellus. Ut vel ullamcorper tellus. Nulla ut mi vitae nunc pellentesque mattis. 
        </h1>
        <blockquote>
          <h1
            class="override"
          >
            Good-looking, powerful, and a sharp handler, it proves you shouldn’t send a crossover out to do a sports sedan’s job.
          </h1>
        </blockquote>
        <h6
          id="unordered-list"
        >
          Unordered List
        </h6>
        <ul>
          <li>
            You can have lists, like this one
          </li>
          <li>
            Make things 
            <strong>
              bold
            </strong>
             or 
            <em>
              italic
            </em>
          </li>
          <li>
            Embed snippets of 
            <code>
              code
            </code>
          </li>
          <li>
            Create 
            <a
              href="/"
            >
              links
            </a>
          </li>
          <li>
            ...
          </li>
        </ul>
        <h6
          id="ordered-list"
        >
          Ordered List
        </h6>
        <ol
          start="1"
        >
          <li>
            You can have lists, like this one
          </li>
          <li>
            Make things 
            <strong>
              bold
            </strong>
             or 
            <em>
              italic
            </em>
          </li>
          <li>
            Embed snippets of 
            <code>
              code
            </code>
          </li>
          <li>
            Create 
            <a
              href="/"
            >
              links
            </a>
          </li>
          <li>
            ...
          </li>
        </ol>
        <h6
          id="another-sub-heading"
        >
          Another Sub Heading
        </h6>
        <h1
          class="override"
        >
          Lorem ip sum dolor sit amet, consectetur adipiscing elit. Nunc scelerisque ex at felis condimentum imperdiet. Proin sodales elit vitae est laoreet, eget vulputate quam pretium. Nulla scelerisque ligula at malesuada mattis. Nulla vulputate posuere orci, id tristique eros pretium nec. Sed a maximus tellus. Ut vel ullamcorper tellus. Nulla ut mi vitae nunc pellentesque mattis.
        </h1>
        <h6
          id="another-sub-heading"
        >
          Another Sub Heading
        </h6>
        <h1
          class="override"
        >
          Lorem ip sum dolor sit amet, consectetur adipiscing elit. Nunc scelerisque ex at felis condimentum imperdiet. Proin sodales elit vitae est laoreet, eget vulputate quam pretium. Nulla scelerisque ligula at malesuada mattis. Nulla vulputate posuere orci, id tristique eros pretium nec. Sed a maximus tellus. Ut vel ullamcorper tellus. Nulla ut mi vitae nunc pellentesque mattis.
        </h1>
      </div>
    </div>
  </div>
</div>
`;

exports[`<FreeFormText> should render with an seoH1 value of "Subtitle" 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 gdTPBx"
    data-gtm-category="Others"
    width="initial"
  >
    <div
      class="Box-sc-1bsf784-0 ezzCAj"
    >
      <div
        class="sc-1k44dfv-0 klHKZs"
      >
        

        <h3
          class="sc-1k44dfv-0 eUGlGC"
        >
          Title
        </h3>
      </div>
    </div>
    <div
      class="Box-sc-1bsf784-0 iPRsIc"
    >
      <h1
        class="sc-1k44dfv-0 eeeuqx"
      >
        Subtitle
      </h1>
      <div
        class="sc-1k44dfv-0 klHKZs sc-bdVaJa bIYKxT"
      >
        

        <p>
          Lorem ip sum 
          <sup>
            dolor sit
          </sup>
           amet, consectetur adipiscing elit. Nunc scelerisque ex at felis condimentum imperdiet. Proin sodales elit vitae est laoreet, eget vulputate quam pretium. Nulla scelerisque ligula at malesuada mattis. Nulla vulputate posuere orci, id tristique eros pretium nec. Sed a maximus tellus. Ut vel ullamcorper tellus. Nulla ut mi vitae nunc pellentesque mattis. 
        </p>
        <blockquote>
          <p>
            Good-looking, powerful, and a sharp handler, it proves you shouldn’t send a crossover out to do a sports sedan’s job.
          </p>
        </blockquote>
        <h6
          id="unordered-list"
        >
          Unordered List
        </h6>
        <ul>
          <li>
            You can have lists, like this one
          </li>
          <li>
            Make things 
            <strong>
              bold
            </strong>
             or 
            <em>
              italic
            </em>
          </li>
          <li>
            Embed snippets of 
            <code>
              code
            </code>
          </li>
          <li>
            Create 
            <a
              href="/"
            >
              links
            </a>
          </li>
          <li>
            ...
          </li>
        </ul>
        <h6
          id="ordered-list"
        >
          Ordered List
        </h6>
        <ol
          start="1"
        >
          <li>
            You can have lists, like this one
          </li>
          <li>
            Make things 
            <strong>
              bold
            </strong>
             or 
            <em>
              italic
            </em>
          </li>
          <li>
            Embed snippets of 
            <code>
              code
            </code>
          </li>
          <li>
            Create 
            <a
              href="/"
            >
              links
            </a>
          </li>
          <li>
            ...
          </li>
        </ol>
        <h6
          id="another-sub-heading"
        >
          Another Sub Heading
        </h6>
        <p>
          Lorem ip sum dolor sit amet, consectetur adipiscing elit. Nunc scelerisque ex at felis condimentum imperdiet. Proin sodales elit vitae est laoreet, eget vulputate quam pretium. Nulla scelerisque ligula at malesuada mattis. Nulla vulputate posuere orci, id tristique eros pretium nec. Sed a maximus tellus. Ut vel ullamcorper tellus. Nulla ut mi vitae nunc pellentesque mattis.
        </p>
        <h6
          id="another-sub-heading"
        >
          Another Sub Heading
        </h6>
        <p>
          Lorem ip sum dolor sit amet, consectetur adipiscing elit. Nunc scelerisque ex at felis condimentum imperdiet. Proin sodales elit vitae est laoreet, eget vulputate quam pretium. Nulla scelerisque ligula at malesuada mattis. Nulla vulputate posuere orci, id tristique eros pretium nec. Sed a maximus tellus. Ut vel ullamcorper tellus. Nulla ut mi vitae nunc pellentesque mattis.
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<FreeFormText> should render with center-aligned content 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 hmddkH"
    data-gtm-category="Others"
    width="initial"
  >
    <div
      class="Box-sc-1bsf784-0 ezzCAj"
    >
      <div
        class="sc-1k44dfv-0 klHKZs"
      >
        

        <h3
          class="sc-1k44dfv-0 eUGlGC"
        >
          Title
        </h3>
      </div>
    </div>
    <div
      class="Box-sc-1bsf784-0 iPRsIc"
    >
      <h6
        class="sc-1k44dfv-0 eeeuqx"
      >
        Subtitle
      </h6>
      <div
        class="sc-1k44dfv-0 klHKZs sc-bdVaJa bIYKxT"
      >
        

        <p>
          Lorem ip sum 
          <sup>
            dolor sit
          </sup>
           amet, consectetur adipiscing elit. Nunc scelerisque ex at felis condimentum imperdiet. Proin sodales elit vitae est laoreet, eget vulputate quam pretium. Nulla scelerisque ligula at malesuada mattis. Nulla vulputate posuere orci, id tristique eros pretium nec. Sed a maximus tellus. Ut vel ullamcorper tellus. Nulla ut mi vitae nunc pellentesque mattis. 
        </p>
        <blockquote>
          <p>
            Good-looking, powerful, and a sharp handler, it proves you shouldn’t send a crossover out to do a sports sedan’s job.
          </p>
        </blockquote>
        <h6
          id="unordered-list"
        >
          Unordered List
        </h6>
        <ul>
          <li>
            You can have lists, like this one
          </li>
          <li>
            Make things 
            <strong>
              bold
            </strong>
             or 
            <em>
              italic
            </em>
          </li>
          <li>
            Embed snippets of 
            <code>
              code
            </code>
          </li>
          <li>
            Create 
            <a
              href="/"
            >
              links
            </a>
          </li>
          <li>
            ...
          </li>
        </ul>
        <h6
          id="ordered-list"
        >
          Ordered List
        </h6>
        <ol
          start="1"
        >
          <li>
            You can have lists, like this one
          </li>
          <li>
            Make things 
            <strong>
              bold
            </strong>
             or 
            <em>
              italic
            </em>
          </li>
          <li>
            Embed snippets of 
            <code>
              code
            </code>
          </li>
          <li>
            Create 
            <a
              href="/"
            >
              links
            </a>
          </li>
          <li>
            ...
          </li>
        </ol>
        <h6
          id="another-sub-heading"
        >
          Another Sub Heading
        </h6>
        <p>
          Lorem ip sum dolor sit amet, consectetur adipiscing elit. Nunc scelerisque ex at felis condimentum imperdiet. Proin sodales elit vitae est laoreet, eget vulputate quam pretium. Nulla scelerisque ligula at malesuada mattis. Nulla vulputate posuere orci, id tristique eros pretium nec. Sed a maximus tellus. Ut vel ullamcorper tellus. Nulla ut mi vitae nunc pellentesque mattis.
        </p>
        <h6
          id="another-sub-heading"
        >
          Another Sub Heading
        </h6>
        <p>
          Lorem ip sum dolor sit amet, consectetur adipiscing elit. Nunc scelerisque ex at felis condimentum imperdiet. Proin sodales elit vitae est laoreet, eget vulputate quam pretium. Nulla scelerisque ligula at malesuada mattis. Nulla vulputate posuere orci, id tristique eros pretium nec. Sed a maximus tellus. Ut vel ullamcorper tellus. Nulla ut mi vitae nunc pellentesque mattis.
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<FreeFormText> should render without errors 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 gdTPBx"
    data-gtm-category="Others"
    width="initial"
  >
    <div
      class="Box-sc-1bsf784-0 ezzCAj"
    >
      <div
        class="sc-1k44dfv-0 klHKZs"
      >
        

        <h3
          class="sc-1k44dfv-0 eUGlGC"
        >
          Title
        </h3>
      </div>
    </div>
    <div
      class="Box-sc-1bsf784-0 iPRsIc"
    >
      <h6
        class="sc-1k44dfv-0 eeeuqx"
      >
        Subtitle
      </h6>
      <div
        class="sc-1k44dfv-0 klHKZs sc-bdVaJa bIYKxT"
      >
        

        <p>
          Lorem ip sum 
          <sup>
            dolor sit
          </sup>
           amet, consectetur adipiscing elit. Nunc scelerisque ex at felis condimentum imperdiet. Proin sodales elit vitae est laoreet, eget vulputate quam pretium. Nulla scelerisque ligula at malesuada mattis. Nulla vulputate posuere orci, id tristique eros pretium nec. Sed a maximus tellus. Ut vel ullamcorper tellus. Nulla ut mi vitae nunc pellentesque mattis. 
        </p>
        <blockquote>
          <p>
            Good-looking, powerful, and a sharp handler, it proves you shouldn’t send a crossover out to do a sports sedan’s job.
          </p>
        </blockquote>
        <h6
          id="unordered-list"
        >
          Unordered List
        </h6>
        <ul>
          <li>
            You can have lists, like this one
          </li>
          <li>
            Make things 
            <strong>
              bold
            </strong>
             or 
            <em>
              italic
            </em>
          </li>
          <li>
            Embed snippets of 
            <code>
              code
            </code>
          </li>
          <li>
            Create 
            <a
              href="/"
            >
              links
            </a>
          </li>
          <li>
            ...
          </li>
        </ul>
        <h6
          id="ordered-list"
        >
          Ordered List
        </h6>
        <ol
          start="1"
        >
          <li>
            You can have lists, like this one
          </li>
          <li>
            Make things 
            <strong>
              bold
            </strong>
             or 
            <em>
              italic
            </em>
          </li>
          <li>
            Embed snippets of 
            <code>
              code
            </code>
          </li>
          <li>
            Create 
            <a
              href="/"
            >
              links
            </a>
          </li>
          <li>
            ...
          </li>
        </ol>
        <h6
          id="another-sub-heading"
        >
          Another Sub Heading
        </h6>
        <p>
          Lorem ip sum dolor sit amet, consectetur adipiscing elit. Nunc scelerisque ex at felis condimentum imperdiet. Proin sodales elit vitae est laoreet, eget vulputate quam pretium. Nulla scelerisque ligula at malesuada mattis. Nulla vulputate posuere orci, id tristique eros pretium nec. Sed a maximus tellus. Ut vel ullamcorper tellus. Nulla ut mi vitae nunc pellentesque mattis.
        </p>
        <h6
          id="another-sub-heading"
        >
          Another Sub Heading
        </h6>
        <p>
          Lorem ip sum dolor sit amet, consectetur adipiscing elit. Nunc scelerisque ex at felis condimentum imperdiet. Proin sodales elit vitae est laoreet, eget vulputate quam pretium. Nulla scelerisque ligula at malesuada mattis. Nulla vulputate posuere orci, id tristique eros pretium nec. Sed a maximus tellus. Ut vel ullamcorper tellus. Nulla ut mi vitae nunc pellentesque mattis.
        </p>
      </div>
    </div>
  </div>
</div>
`;
